### PR TITLE
Made the ui bind to properties earlier in the setup

### PR DESCRIPTION
**Pull request by martint-unity**
### Purpose of this PR
We need to bind these properties earlier during CreateGUI call. Otherwise we had an issue in where we are using the Search API and the UI didn't update properly when the Search was done.
So added an Action to be called after initialization is done for the converters.

image

### CODE REVIEW

**Review by erikabar — APPROVED** [review]
LGTM

File: com.unity.render-pipelines.universal/Editor/Converter/RenderPipelineConvertersEditor.cs [hunk 1]
```diff
@@ -210,144 +283,67 @@ void GetAndSetData(int i)
...
-                Status status;
-                bool active = true;
-                // If this data hasn't been filled in from the init phase then we can assume that there are no issues / warnings
-                if (string.IsNullOrEmpty(converterItemInfos[j].warningMessage))
-                {
-                    status = Status.Pending;
-                }
-                else
+            conv.OnInitialize(initCtx,
+                () =>
```
Comment (svens-unity) [inline com.unity.render-pipelines.universal/Editor/Converter/RenderPipelineConvertersEditor.cs:291]: pref we would move the code out to function.

File: com.unity.render-pipelines.universal/Editor/Converter/RenderPipelineConvertersEditor.cs [hunk 2]
```diff
@@ -210,144 +283,67 @@ void GetAndSetData(int i)
...
-                });
-            }
+                    // Default all the entries to true
+                    for (var j = 0; j < converterItemInfos.Count; j++)
+                    {
+                        string message = string.Empty;
+                        Status status;
+                        bool active = true;
+                        // If this data hasn't been filled in from the init phase then we can assume that there are no issues / warnings
+                        if (string.IsNullOrEmpty(converterItemInfos[j].warningMessage))
```
Comment (svens-unity) [inline com.unity.render-pipelines.universal/Editor/Converter/RenderPipelineConvertersEditor.cs:304]: I dont fully understand the logic here. Having no warning message means the item is still pending? 
We can look into this for a future PR.

**Review by svens-unity — APPROVED** [review]
It would be great to clean up the callbacks and lambdas before expanding usage to other teams or promoting this API to public.

**Review by TheoWong-pixel — APPROVED** [review]

**Review by phi-lira — APPROVED** [review]

**Pull request merged**